cmd/client/noderule.go

```diff
@@ -67,7 +67,10 @@ var NodeRuleCreate = &cobra.Command{
 			os.Exit(1)
 		}
 
-		m, err := usertopology.DefToMetadata(metadata, graph.Metadata{})
+		m, err := usertopology.DefToMetadata(metadata, graph.Metadata{
+			"Name": nodeName,
```

Comment:
while doing update these two fields are optional and we are not going to create any node. so we need to skip this ```isValidNodeMetadata``` check.